src/vulkan/descriptor.h

```diff
@@ -36,31 +36,29 @@ enum class DescriptorType : uint8_t {
   kUniformBuffer,
   kStorageImage,
   kSampledImage,
-  kCombinedImageSampler
+  kCombinedImageSampler,
+  kSampler
 };
 
 class Descriptor {
  public:
-  Descriptor(Buffer* buffer,
-             DescriptorType type,
+  Descriptor(DescriptorType type,
              Device* device,
              uint32_t desc_set,
              uint32_t binding);
   virtual ~Descriptor();
 
+  virtual void UpdateDescriptorSetIfNeeded(VkDescriptorSet descriptor_set) = 0;
+  virtual Result CreateResourceIfNeeded() { return {}; }
```

Comment:
Why was this default'd and not left pure virtual?

Reply:
Maybe it was failing some CI task? Let me try with pure virtual and see what happens.

Reply:
Pure virtual seems to work.